<?xml version="1.0"?>
<sdf version="1.6">
  <model name="my_lrauv">
    <!-- Body -->
    <link name="base_link">
      <inertial>
        <mass>147.8671</mass>
        <inertia>
          <ixx>3.000000</ixx>
          <ixy>0</ixy>
          <ixz>0</ixz>
          <iyy>41.980233</iyy>
          <iyz>0</iyz>
          <izz>41.980233</izz>
        </inertia>
      </inertial>
      <!-- Uncomment this block to enable neutral buoyancy. -->
      <collision name="main_body_buoyancy">
        <geometry>
          <box>
            <size>2 0.3 0.2464451666666667</size>
          </box>
        </geometry>
      </collision>
      <visual name="visual">
        <geometry>
          <mesh>
            <uri>meshes/tethys.dae</uri>
            <submesh>
              <name>Body</name>
                <center>false</center>
            </submesh>
          </mesh>
        </geometry>
        <material>
          <diffuse>1.0 1.0 1.0</diffuse>
          <specular>1.0 1.0 1.0</specular>
          <pbr>
            <metal>
              <albedo_map>materials/textures/Tethys_Albedo.png</albedo_map>
              <normal_map>materials/textures/Tethys_Normal.png</normal_map>
              <metalness_map>materials/textures/Tethys_Metalness.png</metalness_map>
              <roughness_map>materials/textures/Tethys_Roughness.png</roughness_map>
            </metal>
          </pbr>
        </material>
      </visual>
    </link>
    <!-- Horizontal fins -->
    <link name="horizontal_fins">
      <pose>1.05 0 0 0 0 0</pose>
      <inertial>
        <pose>0 0 0 1.57 0 0.5</pose>
        <mass>0.2</mass>
        <inertia>
          <ixx>0.0007924568755</ixx>
          <ixy>-0.0000000002674</ixy>
          <ixz>0.0003978158176</ixz>
          <iyy>0.0010546736475</iyy>
          <iyz>-0.0000000006729</iyz>
          <izz>0.0002633558262</izz>
         </inertia>
      </inertial>
      <collision name="collision">
        <geometry>
          <box>
            <size>0.1 0.1 0.02</size>
          </box>
        </geometry>
      </collision>
      <visual name="visual">
        <pose>-1.05 0 0 0 0 0</pose>
        <geometry>
          <mesh>
            <uri>meshes/tethys.dae</uri>
            <submesh>
              <name>Fins_Horizontal</name>
              <center>false</center>
            </submesh>
          </mesh>
        </geometry>
        <material>
          <diffuse>1.0 1.0 1.0</diffuse>
          <specular>1.0 1.0 1.0</specular>
          <pbr>
            <metal>
              <albedo_map>materials/textures/Tethys_Albedo.png</albedo_map>
              <normal_map>materials/textures/Tethys_Normal.png</normal_map>
              <metalness_map>materials/textures/Tethys_Metalness.png</metalness_map>
              <roughness_map>materials/textures/Tethys_Roughness.png</roughness_map>
            </metal>
          </pbr>
        </material>
        </visual>
    </link>
    <!-- Vertical fins -->
    <link name="vertical_fins">
      <pose>1.05 0 0 0 0 0</pose>
      <inertial>
        <mass>0.2</mass>
        <inertia>
          <ixx>0.0007924568755</ixx>
          <ixy>-0.0000000002674</ixy>
          <ixz>0.0003978158176</ixz>
          <iyy>0.0010546736475</iyy>
          <iyz>-0.0000000006729</iyz>
          <izz>0.0002633558262</izz>
        </inertia>
      </inertial>
      <collision name="collision">
        <geometry>
          <box>
            <size>0.1 0.1 0.02</size>
          </box>
        </geometry>
      </collision>
      <visual name="visual">
        <pose>-1.05 0 0 0 0 0</pose>
        <geometry>
          <mesh>
            <uri>meshes/tethys.dae</uri>
            <submesh>
              <name>Fins_vertical</name>
              <center>false</center>
            </submesh>
          </mesh>
        </geometry>
        <material>
          <diffuse>1.0 1.0 1.0</diffuse>
          <specular>1.0 1.0 1.0</specular>
          <pbr>
            <metal>
              <albedo_map>materials/textures/Tethys_Albedo.png</albedo_map>
              <normal_map>materials/textures/Tethys_Normal.png</normal_map>
              <metalness_map>materials/textures/Tethys_Metalness.png</metalness_map>
              <roughness_map>materials/textures/Tethys_Roughness.png</roughness_map>
            </metal>
          </pbr>
        </material>
      </visual>
    </link>
    <!-- Propeller -->
    <link name="propeller">
      <pose>1.43162 0 0 0 0 0</pose>
      <inertial>
        <pose>0 0 0 0 0 0</pose>
        <mass>0.09</mass>
        <inertia>
          <ixx>0.000143971303</ixx>
          <ixy>0.000000000008</ixy>
          <ixz>-0.000000000224</ixz>
          <iyy>0.000140915448</iyy>
          <iyz>-0.000025236433</iyz>
          <izz>0.000033571862</izz>
        </inertia>
      </inertial>
      <collision name="collision">
        <geometry>
          <box>
            <size>0.03 0.1 0.03</size>
          </box>
        </geometry>
      </collision>
      <visual name="visual">
        <pose>-1.43162 0 0 0 0 0</pose>
        <geometry>
          <mesh>
            <uri>meshes/tethys.dae</uri>
            <submesh>
              <name>Prop</name>
              <center>false</center>
            </submesh>
          </mesh>
        </geometry>
        <material>
          <diffuse>1.0 1.0 1.0</diffuse>
          <specular>1.0 1.0 1.0</specular>
          <pbr>
            <metal>
              <albedo_map>materials/textures/Tethys_Albedo.png</albedo_map>
              <normal_map>materials/textures/Tethys_Normal.png</normal_map>
              <metalness_map>materials/textures/Tethys_Metalness.png</metalness_map>
              <roughness_map>materials/textures/Tethys_Roughness.png</roughness_map>
            </metal>
          </pbr>
        </material>
      </visual>
    </link>
    <!-- Joints -->
    <joint name="horizontal_fins_joint" type="revolute">
      <pose>0 0 0 0 0 0</pose>
      <parent>base_link</parent>
      <child>horizontal_fins</child>
      <axis>
        <xyz>0 1 0</xyz>
        <limit>
          <lower>-0.261799</lower>
          <upper>0.261799</upper>
          <effort>-1</effort>
          <velocity>-1</velocity>
        </limit>
      </axis>
    </joint>
    <joint name="vertical_fins_joint" type="revolute">
      <pose>0 0 0 0 0 0</pose>
      <parent>base_link</parent>
      <child>vertical_fins</child>
      <axis>
        <xyz>0 0 1</xyz>
        <limit>
          <lower>-0.261799</lower>
          <upper>0.261799</upper>
          <effort>-1</effort>
          <velocity>-1</velocity>
        </limit>
      </axis>
    </joint>
    <joint name="propeller_joint" type="revolute">
      <pose>0 0 0 0 0 0</pose>
      <parent>base_link</parent>
      <child>propeller</child>
      <axis>
        <xyz>1 0 0</xyz>
        <limit>
          <lower>-1e+12</lower>
          <upper>1e+12</upper>
          <effort>-1</effort>
          <velocity>-1</velocity>
        </limit>
      </axis>
    </joint>

  </model>
</sdf>
